public/js/post/post.js

```diff
@@ -109,11 +141,15 @@ var MultipleCourseCode = React.createClass({
 
         return (
             <div id={this.props.courseID} className={classes}>
-                <p className="code" onClick={this.toggleFullInfo}> {this.props.data.categoryName} </p>
+                <p className="code" onClick={this.toggleFullInfo}> {this.props.categoryName} </p>
                 <div id = {'spec' + this.props.courseID.substring(5, this.props.courseID.length)} className={infoClasses}>
                     <p className="full_name"> 
-                        {Array.apply(0, Array(this.props.data.textBoxNumber)).map(function (x, i) {
-                            return <input type='text' defaultValue={me.props.data.courses[i]} onKeyDown={me.handleKeyDown} />;
+                        {Array.apply(0, Array(this.props.textBoxNumber)).map(function (x, i) {
+                            if (me.props.textboxesDisabled) {
+                                return <input type='text' defaultValue={me.props.courses[i]} onKeyDown={me.handleKeyDown} disabled />;
```

Comment:
You're right -- the HTML5 spec does say that. It's important to remember that JSX has some differences from HTML's syntax though, and this is actually one of those things (as mentioned in that issue). It'd certainly be cleaner to go with JSX's behaviour.